system/jlib/jlog.hpp

```diff
@@ -829,7 +829,51 @@ constexpr LogMsgCategory MCauditInfo(MSGAUD_audit, MSGCLS_information, AudMsgThr
 constexpr LogMsgCategory MCstats(MSGAUD_operator, MSGCLS_progress, ProgressMsgThreshold);
 constexpr LogMsgCategory MCoperatorInfo(MSGAUD_operator, MSGCLS_information, InfoMsgThreshold);
 
-inline LogMsgCategory MCexception(IException * e, LogMsgClass cls = MSGCLS_error) { return LogMsgCategory((e)->errorAudience(),cls); }
+/*
+ * Function to determine log level (detail) for exceptions, based on log message class
+ */
+inline int mapClassToDefaultDetailLevel(LogMsgClass cls)
+{
+	switch (cls)
+	{
+		case MSGCLS_disaster:
+		case MSGCLS_all:
+			return FatalMsgThreshold;
+		case MSGCLS_error:
+			return ErrMsgThreshold;
+		case MSGCLS_warning:
+			return WarnMsgThreshold;
+		case MSGCLS_information:
+			return InfoMsgThreshold;
+		case MSGCLS_progress:
+			return ProgressMsgThreshold;
+		default:
+			return DefaultDetail;
+	}
+}
+
+/*
```

Comment:
Curious why this needs a different function.  If an exception is reported with class information (I don't think any are), why would it be given a warning level of detail?

Reply:
There are a few instances of MCexception created w/ a WarningClass, and since callers can specify any class (via constructor or other functions such as LogMsgPrepender::report JLog.cpp L:1384) this was meant as a policy to ensure appropriate levels assigned to MCexcpetion. It prob makes more sense to default to ERRthreshold...

Reply:
@rpastrana 
I'm still not sure why an exception must have a minimum level of a warning.  If the class did happen to be information (I don't see any examples) then the detail level should match.  My objection is that it seems like unnecessary code, and I'm not convinced it is correct to force a minimum detail level.

There are two unusual uses of MCexception in daftdir.cpp e.g. 
MCexception(e)(1000)
which will set the detail level to 1000.  The value of the detail level should probably be changed- they might be candidates for an information class, although a warning is probably more appropriate.

There are also several calls to that same function:
    constexpr LogMsgCategory  operator ()(unsigned newDetail) const { return LogMsgCategory(audience, msgClass, newDetail); }
which allow you to override the detail level.  You probably want another PR investigating those and making sure that they fit in with the detail levels you expect.

Reply:
@ghalliday default log level higher than warning doesn't make sense for an exception, but if you insist, we can map the class type to log level directly. 

Although I agree with you, it seems odd to overwrite the log level to 1000, that's a judgement call by the original developer I'm not sure what criteria we could use to change those values.